data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/RemotePeerForwarder.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.peerforwarder;
+
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.peerforwarder.discovery.StaticPeerListProvider;
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import org.opensearch.dataprepper.peerforwarder.client.PeerForwarderClient;
+
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+class RemotePeerForwarder implements PeerForwarder {
+    private final PeerForwarderClientFactory peerForwarderClientFactory;
+    private final PeerForwarderClient peerForwarderClient;
+    private final HashRing hashRing;
+    private final String pipelineName;
+    private final String pluginId;
+    private final Set<String> identificationKeys;
+
+    RemotePeerForwarder(final PeerForwarderClientFactory peerForwarderClientFactory,
+                        final PeerForwarderClient peerForwarderClient,
+                        final HashRing hashRing,
+                        final String pipelineName,
+                        final String pluginId,
+                        final Set<String> identificationKeys) {
+        this.peerForwarderClientFactory = peerForwarderClientFactory;
```

Comment:
We don't use this in this class, this is originally used to create hash ring. Since it's moved to `PeerForwarderProvider` we can remove this

Reply:
Good call. I can remove that.